File: cloud-storage-spring-java/src/main/java/com/github/sevtech/cloud/storage/spring/config/AzureBlobStorageConfig.java
package com.github.sevtech.cloud.storage.spring.config;

import com.azure.storage.blob.BlobServiceClient;
import com.azure.storage.blob.BlobServiceClientBuilder;
import com.github.sevtech.cloud.storage.spring.property.AzureBlobStorageProperties;
import com.github.sevtech.cloud.storage.spring.service.StorageService;
import com.github.sevtech.cloud.storage.spring.service.impl.AzureBlobStorageService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.env.Environment;

@Slf4j
@Configuration
@ConditionalOnCloudStorageProperty(value = "azure.blob.storage.enabled")
public class AzureBlobStorageConfig {

    @Bean
    public AzureBlobStorageProperties azureBlobStorageProperties(final Environment env) {
        return new AzureBlobStorageProperties(env);
    }

    @Bean
    public BlobServiceClient blobServiceClient(final AzureBlobStorageProperties azureBlobStorageProperties) {
        log.info("Registering Azure Blob Storage client");
        return new BlobServiceClientBuilder().connectionString(azureBlobStorageProperties.getConnectionString()).buildClient();
    }

    @Bean
    public StorageService azureBlobStorageService(final BlobServiceClient blobServiceClient) {
        return new AzureBlobStorageService(blobServiceClient);
    }

}

File: cloud-storage-spring-kotlin/src/main/kotlin/com/github/sevtech/cloud/storage/spring/config/AzureBlobStorageConfig.kt
package com.github.sevtech.cloud.storage.spring.config

import com.azure.storage.blob.BlobServiceClient
import com.azure.storage.blob.BlobServiceClientBuilder
import com.github.sevtech.cloud.storage.spring.property.AzureBlobStorageProperties
import com.github.sevtech.cloud.storage.spring.service.StorageService
import com.github.sevtech.cloud.storage.spring.service.impl.AzureBlobStorageService
import mu.KotlinLogging
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.env.Environment

@Configuration
@ConditionalOnCloudStorageProperty(value = "azure.blob.storage.enabled")
class AzureBlobStorageConfig {

    private val log = KotlinLogging.logger {}

    @Bean
    fun azureBlobStorageProperties(env: Environment): AzureBlobStorageProperties {
        return AzureBlobStorageProperties(env)
    }

    @Bean
    fun blobServiceClient(azureBlobStorageProperties: AzureBlobStorageProperties): BlobServiceClient {
        log.info("Registering Azure Blob Storage client")
        return BlobServiceClientBuilder().connectionString(azureBlobStorageProperties.connectionString).buildClient()
    }

    @Bean
    fun azureBlobStorageService(blobServiceClient: BlobServiceClient): StorageService {
        return AzureBlobStorageService(blobServiceClient)
    }
}